### PR TITLE
[android] - build aar only once for release builds

### LINKED ISSUE #8293: Don't build/publish aar if c++ code fails to compile
It seems that `make apackage` builds an aar file even if building of the c++ code failed. [example](https://www.bitrise.io/build/3314ded43a051536).

screen shot 2017-03-06 at 11 56 51

### PR DESCRIPTION
**Pull request by tobrun**
Closes #8293

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -636,7 +636,7 @@ run-android-ui-test-spoon:
 
 .PHONY: apackage
 apackage:
-	cd platform/android && ./gradlew --parallel --max-workers=$(JOBS) assemble$(BUILDTYPE)
+	if [ -d $(“platform/android/MapboxGLAndroidSDK/src/main/jniLibs”) ]; then cd platform/android && ./gradlew --parallel --max-workers=$(JOBS) assemble$(BUILDTYPE); fi
```
Comment (kkaefer) [inline Makefile]: Wrong quotes

Comment (kkaefer) [inline Makefile]: The existence of this directory is not an indication of whether the build was successful or not. The test would succeed on subsequent builds, even if they failed.

Comment (tobrun) [inline Makefile]: This fix was mostly for CI integration where we always start with a clean docker image. Any ideas how to achieve this more cleanly?

Comment (tobrun) [inline Makefile]: I will try getting `shell commands to note the error and exit` to work

**Pull request merged**